pennylane/gradients/gradient_transform.py

```diff
@@ -144,8 +164,32 @@ def jacobian_wrapper(*args, **kwargs):
                 # is present inside the QNode.
                 return qjac
 
-            # Classical processing of a single argument is present. Return qjac @ cjac.
-            jac = qml.math.squeeze(qml.math.tensordot(qml.math.T(cjac), qjac, [[-1], [-1]]))
-            return qml.math.T(jac)
+            # Classical processing present of either:
+            #   a) a single argument
+            #   b) multiple arguments of the same shape
+            # The shape of the classical jacobian returned by qml.jacobian depends on the scenario:
+            #   a) (# gate args, qnode arg shape)
+            #   b) (reverse qnode arg shape, # gate args, # qnode args)
+            # It then needs to be contracted with the quantum jacobian of shape:
+            #      (qnode output shape, # gate args)
+            # The result should have shape:
+            #   a) (qnode ouput shape, qnode arg shape)
+            #   b) (reverse qnode arg shape, qnode output shape, # qnode args)

```

Comment:
It is super nice to have this written down rigorously! :1st_place_medal:

Reply:
Helps me during the implementation as well :)